internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -92,7 +92,7 @@ func (f *FirstSchedulableWithinLimit) FindCluster(kafka *dbapi.KafkaRequest) (*a
 
 	kafkaInstanceSize, e := f.KafkaConfig.GetKafkaInstanceSize(kafka.InstanceType, kafka.SizeId)
 	if e != nil {
-		return nil, e
+		return nil, errors.NewWithCause(errors.ErrorInstancePlanNotSupported, err, "failed to find cluster with criteria '%v'", criteria)
 	}
```

Comment:
Not strictly related to this PR, but I believe this is the least expensive check. Wouldn't be better to make this as first check to avoid doing the more expensive ones if the plan is not supported?

Reply:
good spot, agreed this can be done first before doing anything else.